xir/parser.py

```diff
@@ -275,69 +232,74 @@ def application_stmt(self, args):
         }
         return Statement(name, params, wires, **stmt_options)
 
-    def obs_stmt(self, args):
-        """Observable statement. Defined inside an observable definition.
+    @v_args(inline=True)
+    def obs_def(self, name, params, wires, statements):
 
-        Returns:
-            ObservableStmt: object containing statement data
-        """
-        pref = simplify_math(args[0])
-        terms = args[1]
-        return ObservableStmt(pref, terms, use_floats=self.use_floats)
+        if wires is None:
+            max_wire = 0
+            for stmt in statements:
+                for factor in stmt.factors:
+                    int_wires = [w for w in factor.wires if isinstance(w, int)]
+                    if int_wires and max(int_wires) > max_wire:
+                        max_wire = max(int_wires)
```

Comment:
The `if` statement is redundant here:
```suggestion
                    max_wire = max(max_wire, *int_wires)
```

Reply:
hmm not sure that it is. This is actually copied code from the gate case

Reply:
Hmm, have you checked if the tests pass without the `if` statement? 🤔

Reply:
Oh, true! I suppose there are a couple of workarounds:
```suggestion
                    max_wire = max(max_wire, max_wire, *int_wires)
```
or
```suggestion
                    max_wire = max((max_wire, *int_wires))
```